test/unit/BondingManager.js

```diff
@@ -32,742 +47,1417 @@ contract("BondingManager", accounts => {
+            await bondingManager.bond(1000, transcoder, {from: delegator})
 
-            // Transcoder bonds
-            await bondingManager.bond(2000, tAddr, {from: tAddr})
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 1)
+            await bondingManager.unbond({from: delegator})
+        })
+
+        it("should fail if system is paused", async () => {
+            await fixture.controller.pause()
 
-            // Register transcoder
-            await bondingManager.transcoder(blockRewardCut, feeShare, pricePerSegment, {from: tAddr})
+            await expectThrow(bondingManager.withdrawStake({from: delegator}))
+        })
 
-            // Delegator bonds to transcoder
-            await bondingManager.bond(2000, tAddr, {from: dAddr})
+        it("should fail if current round is not initialized", async () => {
+            await fixture.roundsManager.setMockBool(functionSig("currentRoundInitialized()"), false)
+
+            await expectThrow(bondingManager.withdrawStake({from: delegator}))
+        })
 
-            await fixture.jobsManager.setDistributeFeesParams(tAddr, fees, jobCreationRound)
+        it("should fail if caller is not unbonded", async () => {
+            await bondingManager.bond(100, transcoder, {from: delegator})
 
-            // Set active transcoders
-            await fixture.roundsManager.setCurrentRound(jobCreationRound)
-            await fixture.roundsManager.initializeRound()
-            await fixture.roundsManager.setCurrentRound(currentRound)
+            // This should fail because caller is now in the pending state
+            await expectThrow(bondingManager.withdrawStake({from: delegator}))
         })
 
-        it("should add fees to the fee pool for a round", async () => {
-            // Delegator unbonds and claims fee pool share before fees are distributed
-            await bondingManager.unbond({from: dAddr})
-            // Call updateTranscoderWithFees via transaction from JobsManager
-            await fixture.jobsManager.distributeFees()
+        it("should withdraw caller's bonded stake", async () => {
+            const unbondingPeriod = await bondingManager.unbondingPeriod.call()
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 1 + unbondingPeriod.toNumber())
+            await bondingManager.withdrawStake({from: delegator})
 
-            const expFeePool = fees
-            const tokenPools = await bondingManager.getTranscoderTokenPoolsForRound(tAddr, jobCreationRound)
-            const feePool = tokenPools[1]
-            assert.equal(feePool, expFeePool, "transcoder fee pool incorrect")
+            const dInfo = await bondingManager.getDelegator(delegator)
+            assert.equal(dInfo[0], 0, "should set bondedAmount to 0")
+            assert.equal(dInfo[5], 0, "should set withdrawRound to 0")
```

Comment:
So we can't actually check the token balance in these unit tests because the LivepeerToken contract dependency is mocked out such that the mock doesn't keep track of any actual data - instead whenever the function under test relies on a value returned by the dependency, we just tell the mock to return a specific value. The idea was that for each unit test, we only test state changes specific to the contract under test and leave any side effects created by a function call for a contract to be tested in an integration test where the dependencies are not mocked out.